Monehin/pulse-backend
# Ft: Send email invite functionality

## What?
- Add send invitation email feature
- Add retry send email feature in case of failure 
## Why?
- To automate the process of notifying and enabling authorized prospective users' notification and enablement of registration
- Keep trying to resend to ensure the target recipient is notified at some point in time
## How?
- Trigger email dispatch on successful creation of a registration invite
- Include registration link in invite email
- Calculate an ever-increasing delay for a retry to optimize the performance cost
- Set a maximum retry duration beyond which sending attempts should be aborted
## Testing?
- Either on postman or the Strapi admin UI, create an invite using a working email address. POST: _localhost:1337/invites_ for postman with a minimum body {email: 'your email'}
- Check your inbox for an invite from _atlprwanda@gmail.com_
- To test the retry-send-email functionality:
     - Set your environment variable MAXIMUM_RETRY_DURATION to beyond, say, 300000 ie 5 minutes
     - Delete any existing invite to your test email
     - Create an invite to your email
     - Switch on your data after, say, 4 minutes
     - You should receive an invite sent when internet became 'available'
## Screenshots (optional)
## Anything Else?
- Custom validations currently only work for requests from external clients, not the Strapi admin
- The Sendgrid key for the sender in the sample dotenv file is in the DevPulse channel on slack
